muda/deformers/filter.py

```diff
@@ -0,0 +1,538 @@
+
+
+    def __init__(self):
+        """Abstract base class for Filtering Transformations.
+
+        This implements the deformations, but does not manage state.
+        """
+
+        BaseTransformer.__init__(self)
+
+        # Build the annotation mapping 
+        self._register("pitch_contour", self.filter_contour)
+        self._register("pitch_hz|note_hz", self.filter_hz)
+        self._register("pitch_midi|note_midi", self.filter_midi)
+        self._register("pitch_class", self.filter_class)
+
+
+    def states(self,jam):   
+        mudabox = jam.sandbox.muda
+        state = dict( 
+            nyquist=mudabox._audio["sr"]/2 
+        )
+        yield state
+
+    @staticmethod
+    def audio(mudabox, state):
+
+        if state["btype"] == "bandpass":
+            low,high = state["cut_off"]
+            sos = signal.cheby2(
+                state["order"], 
+                state["attenuation"]/2, 
+                [low,high],
+                btype=state["btype"], 
+                output='sos',
+                fs=mudabox._audio["sr"])
+        else:
+            sos = signal.cheby2(
+                state["order"], 
+                state["attenuation"]/2, 
+                state["cut_off"],
+                btype=state["btype"], 
+                output='sos',
+                fs=mudabox._audio["sr"])
+
+        mudabox._audio["y"] = signal.sosfiltfilt(sos, mudabox._audio["y"])
+
+    
+
+    @staticmethod
+    def filter_contour(annotation, state): 
+    #same length after modification
+        for obs in annotation.pop_data():
+            new_freq,voice = checkfreqinband(obs.value["frequency"],state,datatype="hz")
+            annotation.append(
+                time=obs.time,
+                duration=obs.duration,
+                confidence=obs.confidence,
+                value={
+                    "index": obs.value["index"],
+                    "frequency": new_freq,
+                    "voiced": voice,
+                },
+            )
+
+    @staticmethod
+    def filter_hz(annotation, state):
+        #non-existent pitch removed 
+        for obs in annotation.pop_data():
+            new_freq,voice = checkfreqinband(obs.value,state,datatype="hz")
+            if voice:
+                annotation.append(
+                    time=obs.time,
+                    duration=obs.duration,
+                    confidence=obs.confidence,
+                    value=new_freq,
+                )
+
+    @staticmethod
+    def filter_midi(annotation, state):
+         #non-existent pitch removed 
+        for obs in annotation.pop_data():
+            new_midi,voice = checkfreqinband(obs.value,state,datatype="midi")
+            if voice:
+                annotation.append(
+                    time=obs.time,
+                    duration=obs.duration,
+                    confidence=obs.confidence,
+                    value=new_midi,
+                )
+
+    @staticmethod
+    def filter_class(annotation, state):
```

Comment:
I think we should leave pitch_class annotations unchanged.  Pitch classes don't have octave information, so it's impossible to tell what frequency they actually correspond to, and whether or not they lie in the passband of the filter.

Reply:
I checked https://jams.readthedocs.io/en/stable/namespace.html#pitch-class again and seems like the "tonic" and "pitch" keys together did specify octave information?(for example C7 is convertible to a specific frequency in hz)